Bad conversion of string columns from pandas to datatable

```
>>> dt.Frame(pd.DataFrame(['секрет秘密']))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError: Assertion 'offset < strbuf.size()' failed in src/core/py_buffers.cc, line 194
```

_Originally posted by @arnocandel at https://h2oai.slack.com/archives/C0BECF81M/p1593670773387000_

--- a/src/core/py_buffers.cc
+++ b/src/core/py_buffers.cc
@@ -191,7 +191,7 @@ static void try_to_resolve_object_column(Column& col)
       }
     }
 
-    xassert(offset < strbuf.size());
+    xassert(offset <= strbuf.size());
     strbuf.resize(offset);
     col = Column::new_string_column(nrows, std::move(offbuf), std::move(strbuf));
   }

--- a/tests/frame/test-create.py
+++ b/tests/frame/test-create.py
@@ -1019,6 +1019,12 @@ def test_create_from_pandas_categorical(pandas):
                                ["(6.0, 9.0]"] * 3))
 
 
+def test_issue2517(pandas):
+    DT = dt.Frame(pandas.DataFrame(['секрет秘密']))
+    assert_equals(DT, dt.Frame(['секрет秘密'], names=["0"]))
+
+
+
 
 #-------------------------------------------------------------------------------
 # Create from Numpy

